pkg/realizer/healthcheck/healthcheck.go

```diff
@@ -25,14 +25,39 @@ import (
 	"github.com/vmware-tanzu/cartographer/pkg/utils"
 )
 
+func OwnerHealthCondition(resourceStatuses []v1alpha1.ResourceStatus, previousConditions []metav1.Condition) metav1.Condition {
+	var previousHealthCondition []metav1.Condition
+	condition := conditions.ConditionList(previousConditions).ConditionWithType(v1alpha1.ResourceHealthy)
+	if condition != nil {
+		previousHealthCondition = append(previousHealthCondition, *condition)
+	}
+	healthyConditionManager := conditions.NewConditionManager(v1alpha1.ResourcesHealthy, previousHealthCondition)
+
+	for _, resourceStatus := range resourceStatuses {
+		resourceHealthyCondition := conditions.ConditionList(resourceStatus.Conditions).ConditionWithType(v1alpha1.ResourceHealthy)
+		if resourceHealthyCondition != nil {
+			healthyConditionManager.AddPositive(*resourceHealthyCondition)
+		}
+	}
+
+	healthyConditionResult, _ := healthyConditionManager.Finalize()
+	healthyCondition := healthyConditionResult[len(healthyConditionResult)-1]
```

Comment:
nit: I wonder if we should have a method for this instead of relying on grabbing the latest condition 🤔 

indeed, `Finalize()` will always produce a non-empty array, but, here we're hardcoding the assumption that the aggregated condition is the last one - wdyt of method like "AggregatedCondition()" or something like that that returns that?

Reply:
Perhaps, I'm also not sure ConditionManager was intended to be used this way. Will follow up in a future PR.